google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -134,8 +135,14 @@ public File get() {
         new GenerateConfigActionListener(project, "Dockerfile", new Supplier<File>() {
           @Override
           public File get() {
+            File deploymentFile;
+            if(deploymentSource.getType() instanceof UserSpecifiedPathDeploymentSourceType) {
+              deploymentFile = new File(userSpecifiedArtifactFileSelector.getText());
```

Comment:
That is kind of what I expect the DeploymentSource contract is asking our
implementation to do.

On Wed, Mar 30, 2016 at 1:31 PM, Etan Shaul <notifications@github.com>
wrote:

> In
> google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/565#discussion_r57929418>
> :
>
> > @@ -134,8 +135,14 @@ public File get() {
> >          new GenerateConfigActionListener(project, "Dockerfile", new Supplier<File>() {
> >            @Override
> >            public File get() {
> > +            File deploymentFile;
> > +            if(deploymentSource.getType() instanceof UserSpecifiedPathDeploymentSourceType) {
> > +              deploymentFile = new File(userSpecifiedArtifactFileSelector.getText());
>
> What I mean is, we could override getFile and have it return an internal
> state that we manually set whenever a user browses to a filesystem jar/war.
> Do you like that approach better?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/565/files/5400985cc38833fefc1b984c0c794d4dcf798e8b#r57929418>
>